### ISSUE TITLE
java.lang.NoSuchFieldException: No field producerIndex in class Lgll; after minification by ProGua

### ISSUE THREAD

**Issue by j-s-a-m** [conichiGMBH/conichi-android-sdk#8]
The app crashes at two different spots in the app with SDK version 2.11.0-rc1 that were working without problems with the older SDK version 2.11.0-beta1.

The error message is:

```
java.lang.NoSuchFieldException: No field **producerIndex** in class Lgll; (declaration of 'gll' appears in /data/app/com.hrs.android-2/base.apk:classes2.dex)
glm.java, line 115

```
At least one of the two problems seems to be a ProGuard minification problem, like a hardwired "producerIndex" string used with reflection, because the problem does not occur with a debug build.

> SDK version
2.11.0-rc1
(no problems with 2.11.0-beta1, therefore we had to switch back to 2.11.0-beta1)

> Environment (sandbox | staging | dev | production | ...)
production

> Which iPad Merchant app version (if Merchant App has been used)
not relevant

> Related entities ids (venue_id, card's last four, user_id, etc.)
not relevant (account issue)

> Stacktrace

![bildschirmfoto 2017-11-17 um 17 30 31](https://user-images.githubusercontent.com/6809386/32957848-1bd60406-cbbd-11e7-80b8-d31ca962e434.png)

![bildschirmfoto 2017-11-17 um 17 21 58](https://user-images.githubusercontent.com/6809386/32957836-0d491612-cbbd-11e7-9f62-9f419d54335f.png)

> Steps to reproduce
1.)
Revoke all app permissions, disable Bluetooth, return to your Smarthotel settings screen. I hope you can find the correct code spot in the SDK by using the name of the field ("producerIndex") and the line numbers of the obfuscated stack trace.

2.) 
Enter the Smarthotel settings screen after creating a new Smarthotel account. **Set the Incognito mode.**

> Expected behavior
Don't crash.

> Actual behavior
App crashes.

**Comment by j-s-a-m** [comment]
Because of these new bugs, the importance of the old version 2.11.0-beta1 has grown.

Please fix
https://github.com/conichiGMBH/conichi-android-sdk/issues/7
thoroughly, i.e. by reparing the "core-2.11.0-beta1.aar" here
http://artifactory.conichi.com/artifactory/libs-release-local/com/conichi/sdk/core/2.11.0-beta1/

This file currently is still broken. It has a size of only 4.44 KB. We are lucky that we have an older copy of this file in our local maven repository. If we would have to sync again, we would retrieve the broken version.

**Comment by Ch3D** [comment]
@j-s-a-m okay, let me check this, we haven't caught this issue in our production version of the app

**Comment by Ch3D** [comment]
@j-s-a-m I was able to reproduce it, for some reason consumer proguard file doesn't exist in release aar of the sdk

**Comment by Ch3D** [comment]
Fixed in `2.11.0-rc2`
https://github.com/conichiGMBH/android-sdk/commit/4130959f6e615e19c0e7078264f228eff1fc2529
